libgdf/src/transpose.cu

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2018 NVIDIA Corporation.  All rights reserved. */
+
+#include "gdf_table.cuh"
+#include <gdf/gdf.h>
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <memory>
+#include <stdio.h>
+
+gdf_error gdf_transpose(size_t ncols,
+                        gdf_column** in_cols,
+                        gdf_column** out_cols)
+{
+    // Make sure the inputs are not null
+    if( (0 == ncols) 
+        || (nullptr == in_cols) 
+        || (nullptr == out_cols))
+    {
+        return GDF_DATASET_EMPTY;
+    }
+
+    // If there are no rows in the input, return successfully
+    if ( 0 == in_cols[0]->size )
+    {
+        return GDF_SUCCESS;
+    }
+
+    // Wrap the groupby input columns in a gdf_table
+    using size_type = decltype(ncols);
+    size_t out_ncols = in_cols[0]->size;
+    // TODO (dm): this should be const but the member function in gdf_table.cuh isn't const :(
```

Comment:
Which member function do you need to be `const`? Is it `get_packed_row_values`? If so, then it definitely should be made const. That's just an oversight on my part.